src/GeoJsonPages/GeoJsonContentHandler.php

```diff
@@ -34,19 +38,31 @@ protected function fillParserOutput(
 		ParserOutput &$parserOutput
 	) {
 		'@phan-var GeoJsonContent $content';
+		
+		// @see JsonContentHandler -> fillParserOutput
+		if ( $cpoParams->getGenerateHtml() ) {
+			if ( $content->isValid() ) {
+				$text = $content->getData()->getValue();
+				$parserOutput->setText( $content->rootValueTable( $content->getData()->getValue() ) );
+				
+			} else {
+				$error = wfMessage( 'invalid-json-data' )->parse();
+				$parserOutput->setText( $error );
+			}
 
-		if ( !$cpoParams->getGenerateHtml() || !$content->isValid() ) {
-			$parserOutput->setText( '' );
-			return;
+			$parserOutput->addModuleStyles( [ 'mediawiki.content.json' ] );
+		} else {
+			$parserOutput->setText( null );
 		}
 
-		GeoJsonMapPageUi::forExistingPage( $content->beautifyJSON() )
-			->addToOutput( OutputFacade::newFromParserOutput( $parserOutput ) );
+		// @FIXME alternatively decode $this->mText in GeoJsonLegacyContent
+		// to avoid decoding it again in SubObjectBuilder -> getSubObjectsFromGeoJson
+		$text = json_encode( $content->getData()->getValue() );
```

Comment:
Looks like this should be in the below if-block

Reply:
I have fixed this but not sure about the comment below